lib/_stream_readable.js

```diff
@@ -310,7 +310,8 @@ function readableAddChunk(stream, chunk, encoding, addToFront) {
 }
 
 function addChunk(stream, state, chunk, addToFront) {
-  if (state.flowing && state.length === 0 && !state.sync) {
+  if (state.flowing && state.length === 0 && !state.sync &&
+      stream.listenerCount('data') > 0) {
```

Comment:
Can we run the benchmarks? This is a hot path and this could cost us a bit.

Reply:
That’s probably a comment that better fits into https://github.com/nodejs/node/pull/34957, I’ll mark it as resolved here and kick off a streams benchmark run